### PR TITLE
Adding tests for https://github.com/opscode/chef/pull/2688

### PR DESCRIPTION
**Pull request by tyler-ball**
Completing the tests for https://github.com/chef/chef/pull/2688
Fixes https://github.com/chef/chef/issues/2683

\cc @chef/client-core

### CODE REVIEW

**Comment by jdmundrawala** [comment]
LGTM :+1:

**Comment by jdmundrawala** [comment]
I have no idea what's up with appveyor. The errors do no seem related to your changes. I'm going to tell it to try again

**Comment by ranjib** [comment]
:+1:

**Comment by jdmundrawala** [comment]
@tyler-ball rebase and try again

File: spec/unit/guard_interpreter/resource_guard_interpreter_spec.rb [hunk 1]
```diff
@@ -29,28 +29,105 @@
...
+        context "and guard_interpreter is :script" do
+          let(:parent_resource) do
+            parent_resource = Chef::Resource::Script.new("resource", run_context)
+            parent_resource.guard_interpreter(:script)
+            parent_resource
+          end
+
+          it "merges to :code" do
+            expect(command_opts).to receive(:merge).with({:code => "exit 0"}).and_call_original
+            guard_interpreter.evaluate
```
Comment (jdmundrawala) [inline spec/unit/guard_interpreter/resource_guard_interpreter_spec.rb]: I don't think you should be calling valuate without mocking the parts that shell out

File: spec/unit/guard_interpreter/resource_guard_interpreter_spec.rb [hunk 2]
```diff
@@ -29,28 +29,105 @@
...
+        context "and guard_interpreter is :execute" do
+          let(:parent_resource) do
+            parent_resource = Chef::Resource::Script.new("resource", run_context)
+            parent_resource.guard_interpreter(:execute)
+            parent_resource
+          end
+
+          it "merges to :code" do
+            expect(command_opts).to receive(:merge).with({:command => "exit 0"}).and_call_original
+            guard_interpreter.evaluate
```
Comment (jdmundrawala) [inline spec/unit/guard_interpreter/resource_guard_interpreter_spec.rb]: I don't think you should be calling valuate without mocking the parts that shell out

**Pull request merged**